js/private/node-patches/src/fs.ts

```diff
@@ -0,0 +1,671 @@
+    }
+
+    // tslint:disable-next-line:no-any
+    fs.realpathSync = (...args: any[]) => {
+        const str = origRealpathSync(...args)
+        const escapedRoot = isEscape(args[0], str)
+        if (escapedRoot) {
+            // we've escaped a root; lets the file we've resolved is a symlink and see if our
+            // realpath can be mapped back to the root
+            let linkTarget: string
+            try {
+                linkTarget = path.resolve(
+                    path.dirname(args[0]),
+                    origReadlinkSync(args[0])
+                )
+            } catch (e) {
+                if (e.code === 'EINVAL') {
+                    // the path was not a symlink; just return the resolved path in that case
+                    return str
+                }
+                throw e
+            }
+
+            const realPathRoot = path.resolve(
+                args[0],
+                path.relative(linkTarget, str)
+            )
+            if (!isEscape(args[0], realPathRoot, [escapedRoot])) {
+                return realPathRoot
+            } else {
+                // the realpath has no relative equivalent within the root
+                // so return the actual realpath
+                return str
+            }
+        }
+        return str
+    }
+
+    // tslint:disable-next-line:no-any
+    fs.realpathSync.native = (...args: any[]) => {
+        const str = origRealpathSyncNative(...args)
+        const escapedRoot = isEscape(args[0], str)
+        if (escapedRoot) {
+            // we've escaped a root; lets the file we've resolved is a symlink and see if our
+            // realpath can be mapped back to the root
+            let linkTarget: string
+            try {
+                linkTarget = path.resolve(
+                    path.dirname(args[0]),
+                    origReadlinkSync(args[0])
+                )
+            } catch (e) {
+                if (e.code === 'EINVAL') {
+                    // the path was not a symlink; just return the resolved path in that case
+                    return str
+                }
+                throw e
+            }
+
+            const realPathRoot = path.resolve(
+                args[0],
+                path.relative(linkTarget, str)
+            )
+            if (!isEscape(args[0], realPathRoot, [escapedRoot])) {
+                // this real path can be mapped back to a relative equivalent
+                // within the root
+                return realPathRoot
+            } else {
+                // the realpath has no relative equivalent within the root
+                // so return the actual realpath
+                return str
+            }
+        }
+        return str
+    }
+
+    // =========================================================================
+    // fs.readlink
+    // =========================================================================
+
+    // tslint:disable-next-line:no-any
```

Comment:
may as well toss the tslint lines that are no longer doing anything

Reply:
roger roger